### ISSUE TITLE
Create GPX files with Garmin extensions

### ISSUE THREAD

**Issue by shaunmulligan** [shaunmulligan/hypecycle2#3]
```
gpx = gpxpy.gpx.GPX()

# Add TrackPointExtension namespace
gpx.nsmap["gpxtpx"] = "http://www.garmin.com/xmlschemas/TrackPointExtension/v1"

# Create track
gpx_track = gpxpy.gpx.GPXTrack()
gpx.tracks.append(gpx_track)

# Create segment
gpx_segment = gpxpy.gpx.GPXTrackSegment()
gpx_track.segments.append(gpx_segment)

# Create point:
gpx_point = gpxpy.gpx.GPXTrackPoint(2.1234, 5.1234, time=time())
gpx_segment.points.append(gpx_point)

# Create extension (hr)
hr = 123
gpx_extension_hr = ElementTree.fromstring(f"""<gpxtpx:TrackPointExtension xmlns:gpxtpx="http://www.garmin.com/xmlschemas/TrackPointExtension/v1">
    <gpxtpx:hr>{hr}</gpxtpx:hr>
    </gpxtpx:TrackPointExtension>
""")
gpx_point.extensions.append(gpx_extension_hr)

# Print XML
print(gpx.to_xml())
```

**Comment by shaunmulligan** [comment]
```
import gpxpy
import gpxpy.gpx
from datetime import datetime
from xml.etree import ElementTree
import time

def get_point(lat, lon, ele, power, temp, hr, cadence):
    # Create point:
    gpx_point = gpxpy.gpx.GPXTrackPoint(lat, lon, elevation=ele, time=datetime.now())
    gpx_extension_power = ElementTree.fromstring(f"""<power>{power}</power>""")
    gpx_extension_hr = ElementTree.fromstring(f"""<gpxtpx:TrackPointExtension xmlns:gpxtpx="http://www.garmin.com/xmlschemas/TrackPointExtension/v1">
        <gpxtpx:atemp>{temp}</gpxtpx:atemp>
        <gpxtpx:hr>{hr}</gpxtpx:hr>
        <gpxtpx:cad>{cadence}</gpxtpx:cad>
        </gpxtpx:TrackPointExtension>
    """)
    gpx_point.extensions.append(gpx_extension_power)
    gpx_point.extensions.append(gpx_extension_hr)
    return gpx_point

gpx = gpxpy.gpx.GPX()

# Add TrackPointExtension namespace
gpx.nsmap["gpxtpx"] = "http://www.garmin.com/xmlschemas/TrackPointExtension/v1"

# Create track
gpx_track = gpxpy.gpx.GPXTrack()
gpx.tracks.append(gpx_track)

# Create segment
gpx_segment = gpxpy.gpx.GPXTrackSegment()
gpx_track.segments.append(gpx_segment)

gpx_point = get_point(2.1234, 5.1234, 5.5, 200, 20, 80, 80)
gpx_segment.points.append(gpx_point)
time.sleep(5)
gpx_point = get_point(2.1233, 5.123, 10, 210, 21, 80, 80)
gpx_segment.points.append(gpx_point)
time.sleep(5)
gpx_point = get_point(2.1233, 5.123, 100, 210, 21, 80, 80)
gpx_segment.points.append(gpx_point)
time.sleep(5)
gpx_point = get_point(2.1232, 5.123, 11, 230, 20, 80, 80)
gpx_segment.points.append(gpx_point)

# Print XML
print(gpx.to_xml())

print("Distance: ", gpx.length_3d())
moving_data = gpx.get_moving_data()
if moving_data:
    print('Moving time: ', moving_data.moving_time)
    print('Stopped time: ', moving_data.stopped_time)
    print('Max speed: ' , moving_data.max_speed)
    print('Avg speed: ' , (moving_data.moving_distance / moving_data.moving_time) if moving_data.moving_time > 0 else "?")

uphill, downhill = gpx.get_uphill_downhill()
print('Total uphill: ', uphill)
print('Total downhill: ', downhill)

start_time, end_time = gpx.get_time_bounds()
print('Started: ',  start_time)
print('Ended: ', end_time)
```

**Comment by shaunmulligan** [comment]
closed by #27
